bin/s2nc.c

```diff
@@ -186,223 +195,243 @@ int main(int argc, char *const *argv)
         exit(1);
     }
 
+    if (s2n_init() < 0) {
+        print_s2n_error("Error running s2n_init()");
+        exit(1);
+    }
+
     if ((r = getaddrinfo(host, port, &hints, &ai_list)) != 0) {
         fprintf(stderr, "error: %s\n", gai_strerror(r));
         exit(1);
     }
 
-    int connected = 0;
-    for (ai = ai_list; ai != NULL; ai = ai->ai_next) {
-        if ((sockfd = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol)) == -1) {
-            continue;
+    do {
+        int connected = 0;
+        for (ai = ai_list; ai != NULL; ai = ai->ai_next) {
+            if ((sockfd = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol)) == -1) {
+                continue;
+            }
+
+            if (connect(sockfd, ai->ai_addr, ai->ai_addrlen) == -1) {
+                close(sockfd);
+                continue;
+            }
+
+            connected = 1;
+            /* connect() succeeded */
+            break;
         }
 
-        if (connect(sockfd, ai->ai_addr, ai->ai_addrlen) == -1) {
+        if (connected == 0) {
+            fprintf(stderr, "Failed to connect to %s:%s\n", host, port);
             close(sockfd);
```

Comment:
nit: I don't think close is needed here as we already close if connect fails.

Reply:
I just re-arranged this part of code, will remove close call.